autofaiss/external/build.py

```diff
@@ -69,7 +70,9 @@ def estimate_memory_required_for_index_creation(
     else:
         memory_for_training = 0
 
-    return (int(index_overhead + max(index_memory + needed_for_adding, memory_for_training))), index_key
+    index_memory_with_n_indices = index_memory / nb_indices_to_keep
+
+    return (int(index_overhead + max(index_memory_with_n_indices + needed_for_adding, memory_for_training))), index_key
```

Comment:
I think we need to remove/tweak the index_overhead part

Reply:
Do you know why we have set `index_overhead = index_memory * 0.1` ?

Reply:
No good reason, I think we added it because we couldn't explain the additional memory (due to the bug we fixed relatively to items IDs)

The index_memory * 0.25 for training is also wrong.
The size of the index for training is *not* dependent on the total number of embedding.
It's a function of the index parameters, it would be better to compute it properly. 
(Otherwise all terms depending on total number of items will be blockers to compute large indices)

Reply:
I will think about it and tweak these estimations in the next commit.